regions_www/m/server.R

```diff
@@ -253,64 +261,51 @@ shinyServer(function(input, output, session) {
+               region_geo_change_to <- input$geography
+             } else {
+               region_geo_change_to <- "msoa"
+             }
+           },
+           "school"= { region_geo_change_to <- "lsoa" },
+           "alltrips"= { region_geo_change_to <- "msoa" }
+    )
+
     # Only trigger geography changes if required.
-    if (is.na(region$geography) || region_geo_change_to != region$geography){
-      region$geography <<- region_geo_change_to
+    if (!identical(region_geo_change_to, region$geography)) {
+      region$geography <- region_geo_change_to
     }
 
     # Set data_dir
-    region$data_dir <<- file.path(data_regional_root, input_purpose(), region$geography, region$current)
+    region$data_dir <- file.path(data_regional_root, input_purpose(), region$geography, region$current)
 
     # Identify that region repopulation has happened
-    region$repopulate_region <<- T
+    region$repopulate_region <- T
 
     # Identify purposes and geographies available in region
     purposes_list <- c("commute", "school", "alltrips")
-    region$purposes_present <<- (dir.exists(file.path(data_regional_root, purposes_list, "msoa", region$current)) | dir.exists(file.path(data_regional_root, purposes_list, "lsoa", region$current)))
+    new_purpose <- (dir.exists(file.path(data_regional_root, purposes_list, "msoa", region$current)) | dir.exists(file.path(data_regional_root, purposes_list, "lsoa", region$current)))
+    if(!identical(new_purpose,region$purposes_present)){
+      region$purposes_present <- new_purpose
+    }
     geographies_list <- c("msoa", "lsoa")
-    region$geographies_present <<- dir.exists(file.path(data_regional_root, input_purpose(), geographies_list, region$current))
+    region$geographies_present <- dir.exists(file.path(data_regional_root, input_purpose(), geographies_list, region$current))
 
     # Identify the centre of the current region, save in to_plot
     to_plot$center_dim <<- rgeos::gCentroid(regions[regions$region_name == region$current, ], byid = TRUE)@coords
 
     # Load data to to_plot (if data exists - this varies by purpose/geography)
-    if (file.exists(file.path(region$data_dir, "z.Rds"))) {
-      to_plot$zones <<- readRDS(file.path(region$data_dir, "z.Rds"))
-    } else {
-      to_plot$zones <<- NULL
-    }
-
-    if (file.exists(file.path(region$data_dir, "c.Rds"))) {
-      to_plot$centroids <<- readRDS(file.path(region$data_dir,  "c.Rds"))
-    } else {
-      to_plot$centroids <<- NULL
-    }
-
-    if (file.exists(file.path(region$data_dir, "l.Rds"))) {
-      to_plot$straight_lines <<- readRDS(file.path(region$data_dir, "l.Rds"))
-    } else {
-      to_plot$straight_lines <<- NULL
-    }
-
-    if (file.exists(file.path(region$data_dir, "rf.Rds"))) {
-      to_plot$routes_fast <<- readRDS(file.path(region$data_dir, "rf.Rds"))
-    } else {
-      to_plot$routes_fast <<- NULL
+    to_plot$zones <<- load_data(file.path(region$data_dir, "z.Rds"))
```

Comment:
Why `<<-` here but not previously? Suggest `<-` unless it breaks something for consistency.

Reply:
Sadly not.  `to_plot` is not a `reactiveValue` and so global assignment is required.
We can change it but not in this PR please.